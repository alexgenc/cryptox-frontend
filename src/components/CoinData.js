import React from "react";

const CoinData = ({ data }) => {
  const renderData = () => {
    if (data) {
      return (
        <div className="bg-white mt-3 mb-5 p-2 rounded border row">
          <div className="col-sm">
            <div className="d-flex flex-column">
              <span className="font-bold my-2">Market Cap</span>
              <span className="my-3 text-xl">${data.market_cap ? data.market_cap.toLocaleString() : '-'}</span>
            </div>
            <hr />
            <div className="d-flex flex-column">
              <span className="font-bold my-2">
                Total Supply
              </span>
              <span className="my-3 text-xl">{data.total_supply ? data.total_supply.toLocaleString() : '-'}</span>
            </div>
          </div>

          <div className="col-sm">
            <div className="d-flex flex-column">
              <span className="font-bold my-2">Total Volume</span>
              <span className="my-3 text-xl">${data.total_volume ? data.total_volume.toLocaleString() : '-'}</span>
            </div>
            <hr />
            <div className="d-flex flex-column">
              <span className="font-bold my-2">High (24h)</span>
              <span className="my-3 text-xl">${data.high_24h ? data.high_24h.toLocaleString() : '-'}</span>
            </div>
          </div>

          <div className="col-sm">
            <div className="d-flex flex-column">
              <span className="font-bold my-2">
                Circulating Supply
              </span>
              <span className="my-3 text-xl">{data.circulating_supply ? data.circulating_supply.toLocaleString() : '-'}</span>
            </div>
            <hr />
            <div className="d-flex flex-column">
              <span className="font-bold my-2">Low (24h)</span>
              <span className="my-3 text-xl">${data.low_24h ? data.low_24h.toLocaleString() : '-'}</span>
            </div>
          </div>
        </div>
      );
    }
  };

  return <div className="mb-8 mt-3">{renderData()}</div>;
};

export default CoinData;