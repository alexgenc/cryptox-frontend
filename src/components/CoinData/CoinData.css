.coinData {
  margin-bottom: 100px;
  margin-top: 20px;
}

h2 {
  margin-top: 100px;
}