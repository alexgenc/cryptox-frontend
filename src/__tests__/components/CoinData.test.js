import React from 'react';
import { render } from '@testing-library/react';
import CoinData from '../../components/CoinData/CoinData';

const data = {
  "market_cap": 1000000,
  "total_supply": 5000000,
  "total_volume": 300000,
  "high_24h": 500,
  "circulating_supply": 1000000,
  "low_24h": 450
}

describe('<CoinData />', () => {
  // Smoke Test
  it("renders without crashing", () => {
    render(<CoinData data={data} />);
  });

  // Snapshot Test
  it("matches snapshot", () => {
    const {asFragment} = render(<CoinData data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  
  it('renders the <CoinData /> with populated data', () => {
    const { getByText } = render(
      <CoinData data={data} />
    );

    expect(getByText('Market Cap')).toBeTruthy();
    expect(getByText('Total Supply')).toBeTruthy();
    expect(getByText('Total Volume')).toBeTruthy();
    expect(getByText('Circulating Supply')).toBeTruthy();
    expect(getByText('High (24h)')).toBeTruthy();
    expect(getByText('Low (24h)')).toBeTruthy();
  });
});