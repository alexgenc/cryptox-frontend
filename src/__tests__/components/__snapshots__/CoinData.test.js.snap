// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoinData /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="coinData"
  >
    <div
      class="bg-white mt-3 p-2 rounded border row"
    >
      <div
        class="col-sm"
      >
        <div
          class="d-flex flex-column"
        >
          <span
            class="text-muted coin-data-category"
          >
            Market Cap
          </span>
          <span>
            $1,000,000
          </span>
        </div>
        <hr />
        <div
          class="d-flex flex-column"
        >
          <span
            class="text-muted coin-data-category"
          >
            Total Supply
          </span>
          <span>
            5,000,000
          </span>
        </div>
      </div>
      <div
        class="col-sm"
      >
        <div
          class="d-flex flex-column"
        >
          <span
            class="text-muted coin-data-category"
          >
            Total Volume
          </span>
          <span>
            $300,000
          </span>
        </div>
        <hr />
        <div
          class="d-flex flex-column"
        >
          <span
            class="text-muted coin-data-category"
          >
            High (24h)
          </span>
          <span>
            $500
          </span>
        </div>
      </div>
      <div
        class="col-sm"
      >
        <div
          class="d-flex flex-column"
        >
          <span
            class="text-muted coin-data-category"
          >
            Circulating Supply
          </span>
          <span>
            1,000,000
          </span>
        </div>
        <hr />
        <div
          class="d-flex flex-column"
        >
          <span
            class="text-muted coin-data-category"
          >
            Low (24h)
          </span>
          <span>
            $450
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
